File: src/main/java/org/code4everything/hutool/converter/WeekConverter.java
package org.code4everything.hutool.converter;

import cn.hutool.core.date.Week;
import org.code4everything.hutool.Converter;

/**
 * @author pantao
 * @since 2020/10/31
 */
public class WeekConverter implements Converter<Week> {

    @Override
    public Week string2Object(String string) {
        return Week.valueOf(string.toUpperCase());
    }

    @Override
    public String object2String(Object object) {
        return object instanceof Week ? ((Week) object).toChinese() : "";
    }
}

File: src/main/kotlin/org/code4everything/hutool/converter/WeekConverter.kt
package org.code4everything.hutool.converter

import cn.hutool.core.date.Week
import org.code4everything.hutool.Converter

class WeekConverter : Converter<Week> {

    override fun string2Object(string: String): Week = Week.valueOf(string.uppercase())

    override fun object2String(any: Any?): String = if (any is Week) any.toChinese() else ""
}
